crm_exception/models/crm_lead.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class CrmLead(models.Model):
+    _inherit = ["crm.lead", "base.exception"]
+    _name = "crm.lead"
+    _order = "main_exception_id asc, name desc"
+
+    @api.model
+    def _reverse_field(self):
+        return "crm_lead_ids"
+
+    def _fields_trigger_check_exception(self):
+        return ["ignore_exception", "stage_id"]
+
+    def crm_check_exception(self):
+        self._check_exception()
+
+    def _check_crm_lead_check_exception(self, vals):
+        check_exceptions = any(
+            field in vals for field in self._fields_trigger_check_exception()
+        )
+        if check_exceptions:
+            self.crm_check_exception()
+
+    def write(self, vals):
```

Comment:
@AungKoKoLin1997 Any reason to use `write()` instead of `@api.constrains()`?  Shouldn't the latter make the code more concise?

Reply:
The main purpose is to avoid hardcoding a trigger field list in @api.constrains and ensure flexibility for potential future additions to the _fields_trigger_check_exception. May be rare case but I think it is also good for design.

Reply:
@AungKoKoLin1997 Isn't that practically the same as the trigger fields are hardcoded anyway in `_fields_trigger_check_exception()` or am I missing something?

Reply:
Yes. The trigger fields are hardcoded in `_fields_trigger_check_exception()` and we may extend this function and can add new more fields from another module if we will need some adjustments in the future.